pkg/plugin/driver/base.go

```diff
@@ -211,3 +216,65 @@ func (b *Base) CreateWorkerContainerDefintion(hostname string, cert *tls.Certifi
 	}
 	return container
 }
+
+// DefaultPodSpec returns the default PodSpec used by DaemonSet plugins
+func defaultDaemonSetPodSpec() v1.PodSpec {
+	podSpec := v1.PodSpec{
+		Containers:         []v1.Container{},
+		DNSPolicy:          v1.DNSClusterFirstWithHostNet,
+		HostIPC:            true,
+		HostPID:            true,
+		HostNetwork:        true,
+		ServiceAccountName: "sonobuoy-serviceaccount",
+		Tolerations: []v1.Toleration{
+			{
+				Operator: v1.TolerationOpExists,
+			},
+		},
+		Volumes: []v1.Volume{
+			{
+				Name: "root",
+				VolumeSource: v1.VolumeSource{
+					HostPath: &v1.HostPathVolumeSource{
+						Path: "/",
+					},
+				},
+			},
+		},
+	}
+	return podSpec
+}
+
+// DefaultPodSpec returns the default PodSpec used by Job plugins
+func defaultJobPodSpec() v1.PodSpec {
+	return v1.PodSpec{
+		Containers:         []v1.Container{},
+		RestartPolicy:      v1.RestartPolicyNever,
+		ServiceAccountName: "sonobuoy-serviceaccount",
+		Tolerations: []v1.Toleration{
+			{
+				Key:      "node-role.kubernetes.io/master",
+				Operator: v1.TolerationOpExists,
+				Effect:   v1.TaintEffectNoSchedule,
+			},
+			{
+				Key:      "CriticalAddonsOnly",
+				Operator: v1.TolerationOpExists,
+			},
+		},
+		Volumes: []v1.Volume{},
+	}
+}
+
+// DefaultPodSpec returns the default pod spec used for the given plugin driver type.
+func DefaultPodSpec(d string) (v1.PodSpec, error) {
+	switch d {
+	case "DaemonSet":
+		return defaultDaemonSetPodSpec(), nil
+	case "Job":
+		return defaultJobPodSpec(), nil
+	default:
+		return v1.PodSpec{}, errors.Errorf("unknown driver %q", d)
```

Comment:
You could also avoid dealing with the error case here at all and just do something like default to the `defaultJobPodSpec` when it is unknown. I think that is a pretty reasonable thing to do instead of blowing up at this stage. Just makes it a sane default.

That would just simplify the signature and call sites of this. Just an idea; not a requirement.

Reply:
:+1: That's a sensible default. Adding the error caused a chain of changes elsewhere that I'd rather avoid.